### PR TITLE
feature/CRC-21

### PR DESCRIPTION
**Pull request by CodeAnnihilator**

### CODE REVIEW

File: src/screens/Settings/components/Settings.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import {Text, View, Switch, Dimensions, KeyboardAvoidingView, ScrollView, Button, Alert} from 'react-native';
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+const styles = EStyleSheet.create({
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/Settings.tsx]: move it to a separate file called settingsStyles.ts

File: src/screens/Settings/components/Settings.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		marginVertical: 20,
+	}
+});
+
+export default class Settings extends React.PureComponent<any> {
+	public state = {
+		isAlertVisible: false,
+	};
+
+	private readonly callAlert = () => {
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/Settings.tsx]: make it a separate component instead in a separate file

File: src/screens/Settings/components/Settings.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+					text: 'Cancel',
+					onPress: () => this.setState({isAlertVisible: false}),
+					style: 'cancel',
+				},
+			],
+			{cancelable: false},
+		);
+	}
+
+	private readonly switchRenderer = (type: string, index: number, arr: string[]) => {
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/Settings.tsx]: put this component in a separate file

File: src/screens/Settings/components/Settings.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		return (
+			<View
+				key={index}
+				style={style}
+			>
+				<Text>{`${type.charAt(0).toUpperCase()}${type.slice(1)}`}</Text>
+				<Switch
+					value={this.props.temperatureScale === type}
+					onValueChange={() => this.props.changeTemperatureScale(type)}
+				/>
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/Settings.tsx]: const {changeTemperatureScale} = this.props

File: src/screens/Settings/components/Settings.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+				enabled
+			>
+				{this.state.isAlertVisible ? this.callAlert() : null}
+				<ScrollView>
+					<View style={styles.user}>
+						<Text style={{fontSize: 24}}>Settings</Text>
+					</View>
+					<View style={styles.settingsBox}>
+						<Text style={styles.settingsBoxHeader}>Temperature scale:</Text>
+						{['fahrenheit', 'celsius'].map(this.switchRenderer)}
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/Settings.tsx]: {'fahrenheit' && <SomeComponent />}
{'celsius' && <SomeOtherComponent />}

File: src/screens/Settings/components/SettingsResetPassword.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import {Text, View, TextInput, KeyboardAvoidingView} from 'react-native';
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+const styles = EStyleSheet.create({
```
Comment (CodeAnnihilator) [inline src/screens/Settings/components/SettingsResetPassword.tsx]: move it to a separate file called settingsResetPasswordStyles.ts

File: src/screens/Settings/components/Settings.tsx (hunk 5 above)
Comment (Gerasimov94) [reply src/screens/Settings/components/Settings.tsx]: why? we must have a all options to switch

File: src/screens/Settings/constants/settingsConstants.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export const SETTINGS_CHANGE_TEMPERATURE_SCALE = '@@SETTINGS/SETTINGS_CHANGE_TEMPERATURE_SCALE';
```
Comment (CodeAnnihilator) [inline src/screens/Settings/constants/settingsConstants.ts:1]: we can call it just like this: export const CHANGE_TEMPERATURE_SCALE = '@@SETTINGS/CHANGE_TEMPERATURE_SCALE' this way we male it more declarative and short. Let's follow this pattern

**Pull request merged**